orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/parallel/WaitForRequisiteCompletionTask.groovy

```diff
@@ -38,27 +37,35 @@ class WaitForRequisiteCompletionTask implements RetryableTask {
   @Override
   TaskResult execute(Stage stage) {
     boolean allRequisiteStagesAreComplete = true
-    Set<String> termainalStageNames = []
+    Set<String> terminalStageNames = []
 
     def requisiteIds = stage.context.requisiteIds as List<String>
     requisiteIds?.each { String requisiteId ->
       def requisiteStage = stage.execution.stages.find { it.refId == requisiteId }
-      if (requisiteStage?.status != SUCCEEDED) {
+      if (!requisiteStage) {
         allRequisiteStagesAreComplete = false
-      }
-      if (requisiteStage?.status == TERMINAL) {
-        termainalStageNames << requisiteStage?.name
+        return
       }
 
-      def tasks = (requisiteStage?.tasks ?: []) as List<Task>
-      if (tasks && tasks[-1].status != SUCCEEDED) {
-        // ensure the last task has completed (heuristic for all tasks being complete)
-        allRequisiteStagesAreComplete = false
+      def requisiteStages = [requisiteStage] + stage.execution.stages.findAll { it.parentStageId == requisiteStage.id }
```

Comment:
This was previously only looking at the requisiteStage itself ... in the case of a Canary this would have been an initialization stage that was almost certainly complete (with synthetic children still running*).

This changes the WaitFor logic to also check that all synthetic children are complete before proceeding. 

* I'd suggest tackling the initialization stage statuses independently from this PR